File: src/my_py_pkg/my_py_pkg/add_2_ints_client_no_oop.py
#!/usr/bin/env python3
#

import rclpy
from rclpy.node import Node

# ros2 interface show example_interfaces/srv/AddTwoInts
# AddTwoInts.a <- name of the first op. 
# AddTwoInts.b <- name of the second op.
# AddTwoInts.sum <- response sum = a + b.
from example_interfaces.srv import AddTwoInts

SERVICE_NAME = "add_2_ints"

def main( args = None ):
    rclpy.init( args = args )
    node = Node( "add_2_ints_client_no_oop" )
    node.get_logger().info( "ADD2IntsClient has been started." )

    client = node.create_client( AddTwoInts, SERVICE_NAME )
    while not client.wait_for_service( 1.0 ):
        node.get_logger().warn( "Waiting for AddTwoInts server." )

    request = AddTwoInts.Request()
    request.a = 3
    request.b = 8

    future = client.call_async( request )
    rclpy.spin_until_future_complete( node, future )
    response: AddTwoInts.Response = future.result()
    node.get_logger().info( str( request.a ) + " + " + 
                            str( request.b ) +
                            " = " + str( response.sum ) )

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/add_2_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
/*
# ros2 interface show example_interfaces/srv/AddTwoInts
# AddTwoInts.a <- name of the first op. 
# AddTwoInts.b <- name of the second op.
# AddTwoInts.sum <- response sum = a + b.
*/
#include "example_interfaces/srv/add_two_ints.hpp"

// using namespace std::chrono_literals;

#define SERVICE_NAME "add_2_ints"

int main( int argc, char **argv )
{
    rclcpp::init( argc, argv );
    auto node = std::make_shared<rclcpp::Node>( "add_2_ints_client_no_oop" );

    auto client = node->create_client<example_interfaces::srv::AddTwoInts>( SERVICE_NAME );
    while( !client->wait_for_service( std::chrono::seconds( 1 ) ) ) {
        RCLCPP_WARN( node->get_logger(), "Waiting for AddTwoInts server." );
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 3;
    request->b = -1;

    auto future = client->async_send_request( request );
    rclcpp::spin_until_future_complete( node, future );

    auto response =  future.get();
    RCLCPP_INFO( node->get_logger(), "%ld + %ld = %ld",
                 request->a, request->b, response->sum );

    rclcpp::shutdown();
    return 0;
};
